app/module/drivers/input/input_pinnacle.c

```diff
@@ -0,0 +1,386 @@
+#define DT_DRV_COMPAT cirque_pinnacle
+
+#include <zephyr/init.h>
+#include <zephyr/input/input.h>
+#include <zephyr/dt-bindings/input/input-event-codes.h>
+
+#include <zephyr/logging/log.h>
+
+#include "input_pinnacle.h"
+
+LOG_MODULE_REGISTER(pinnacle, CONFIG_INPUT_LOG_LEVEL);
+
+static int pinnacle_seq_read(const struct device *dev, const uint8_t addr, uint8_t *buf,
+                             const uint8_t len) {
+    const struct pinnacle_config *config = dev->config;
+#if DT_INST_ON_BUS(0, spi)
```

Comment:
Unsure if the hardcoded `0` works for multiple instances of the same sensor.

Reply:
It does not. To support more than one input, including ones on different buses, will require setting up some function pointers for read/write in the config for each instance like is done for other drivers. Future work, IMHO.